DVOP-751 Evaluate and update GHA

palchyk-alex opened the pull request:
## Create GitHub Action to run build, test and publish.
This PR edits GitHub Action workflow file, for automated RocketChat build, test and publish process.

| **ClickUp** | **https://app.clickup.com/t/14247018/DEVOPS-2192** |
| ---------- | -------------------------------------------------------- |
| **Jira** | **https://drivecentric.atlassian.net/browse/DVOP-751** |

palchyk-alex commented:
Before merging, need to change secrets in this repo

JamesLoftis commented on .github/workflows/build-release.yml:
```diff
@@ -0,0 +1,320 @@
+      with:
+        versionSpec: '5.6.7'
+        
+    - name: Get release version
```
This one is still using gitversion

JamesLoftis commented on .github/workflows/build-release.yml:
```diff
@@ -0,0 +1,320 @@
+        IMAGE="${{ steps.login-ecr.outputs.registry }}/${ECR_REPOSITORY}"
+        cd /tmp/build
+        docker build -t ${IMAGE}:$GIT_TAG .
+        docker push ${IMAGE}:$GIT_TAG
```
Need to add a script to create the octopus release to the end

palchyk-alex commented:
## Changes summary

1. Added GitHub Actions files for releases and for PRs
2. Removed Jenkinsfile

## Release steps

1. Merge this **PR** to `develop`
3. Verify image creation in [MGMT accounts registry](https://us-east-1.console.aws.amazon.com/ecr/repositories/private/476851419095/rocketchat?region=us-east-1)
